### PR TITLE
Use a TypeVar for the return types of TarFile classmethods.

### PR DESCRIPTION
**Pull request by domdfcoding**
The implementations of the classmethods (and ``__enter__``) return the same type as the class, including for subclasses. However, the current stubs show the methods unconditionally returning `TarFile`, and never its subclasses.

### CODE REVIEW

File: stdlib/tarfile.pyi [hunk 1]
```diff
@@ -129,7 +131,7 @@ class TarFile:
     def __iter__(self) -> Iterator[TarInfo]: ...
     @classmethod
     def open(
-        cls,
+        cls: Type[_TF],
```
Comment (srittau) [inline stdlib/tarfile.pyi:134]: Nit: `type[_TF]`

Comment (domdfcoding) [reply stdlib/tarfile.pyi]: mypy doesn't like `type[_TF]` because of https://github.com/python/mypy/issues/10303

Comment (srittau) [reply stdlib/tarfile.pyi]: Sorry about that. I actually tested it before this comment, but it seems that it worked in my small test for some reason, but not here. 😕

Comment (domdfcoding) [reply stdlib/tarfile.pyi:134]: I have switched back to `Type[_TF]` for now.

**Review by Akuli — APPROVED** [review]

**Comment by Akuli** [comment]
Nit for future reference: We generally prefer adding more commits as opposed to force-pushing. When we merge the PR, it's combined into a single commit anyway.

**Pull request merged**